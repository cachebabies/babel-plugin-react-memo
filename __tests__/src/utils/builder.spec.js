import * as babel from 'babel-core';
import {
  updateReactImportNode,
  arrowFunctionExpressionVisitor,
} from '../../../src/utils/builder';
import {
  importPlugin,
  arrowFunctionPlugin,
} from '../../fixtures/plugins';

let program;

describe('utils', () => {
  describe('updateReactImportNode', () => {
    it('should add named export to react default export', () => {
      program = 'import React from \'react\'';
      const plugin = importPlugin(updateReactImportNode);
      const { code } = babel.transform(program, { plugins: [plugin] });
      expect(code).toMatchSnapshot();
    });

    it('should extend named exports', () => {
      program = 'import React, { Component, Foo } from \'react\'';
      const plugin = importPlugin(updateReactImportNode);
      const { code } = babel.transform(program, { plugins: [plugin] });
      expect(code).toMatchSnapshot();
    });

    it('should not extend named exports if not react', () => {
      program = 'import { isEqual } from \'lodash\'';
      const plugin = importPlugin(updateReactImportNode);
      const { code } = babel.transform(program, { plugins: [plugin] });
      expect(code).toMatchSnapshot();
    });
  });

  describe('arrowFunctionExpressionVisitor', () => {
    it('should wrap arrow function with memo', () => {
      program = 'const fakeComponent = () => { return (<div>hello</div>); }';
      const plugin = arrowFunctionPlugin(arrowFunctionExpressionVisitor);
      const { code } = babel.transform(program, {
        plugins: [plugin, 'syntax-jsx'],
      });
      expect(code).toMatchSnapshot();
    });
  });
});
