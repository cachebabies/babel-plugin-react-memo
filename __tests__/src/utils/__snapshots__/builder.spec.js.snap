// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils updateFunctionalComponent should update the right way 1`] = `
"const fakeComponent = () => {
  return <div>hello</div>;
};"
`;

exports[`utils updateFunctionalComponent should wrap variable declaration with memo 1`] = `
"const fakeComponent = () => {
  return <div>hello</div>;
};"
`;

exports[`utils updateReactImportNode should add named export to react default export 1`] = `"import React, { memo } from 'react';"`;

exports[`utils updateReactImportNode should extend named exports 1`] = `"import React, { Component, Foo, memo } from 'react';"`;
